src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,103 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private Node<K, V>[] buckets;
+
+    private int size;
+    private int capacity = DEFAULT_CAPACITY;
+    private double threshold = capacity * LOAD_FACTOR;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        Node<K, V> newNode = new Node<>(key, value);
+        Node<K, V> currentBucket = buckets[getIndex(key)];
+        int index = getIndex(key);
+
+        if (currentBucket != null) {
+            while (currentBucket != null) {
+                if ((currentBucket.key != null && currentBucket.key.equals(key))
+                        || (currentBucket.key == null && key == null)) {
+                    currentBucket.value = value;
+                    return;
+                }
+                currentBucket = currentBucket.next;
+            }
+            newNode.next = buckets[index];
+            buckets[index] = newNode;
 
+        } else {
+            buckets[index] = newNode;
+        }
+        size++;
+        if (size >= threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> current = buckets[index];
+        while (current != null) {
+            if (Objects.equals(current.key, key)) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        if (key == null) {
+            return 0;
+        }
+        int hash = key.hashCode();
+        return Math.abs(hash) % capacity;
     }
+
+    private void resize() {
+        final int newCapacity = capacity * 2;
```

Comment:
You've marked my comment as resolved but forgot to make multiplier a constant.

Reply:
Thats right, I thought I need to make the newCapacity a final number instead of creating a final number for the multiplier. Fixed.